### PR TITLE
Update smoke test environment

### PR DESCRIPTION
**Pull request by rfratto**
- update to k8s 1.21
- make smoke test bash script work on macOS

### CODE REVIEW

**Comment by rfratto** [comment]
@mattdurham @rlankfo The updates to vendor aren't going to make this one easy to review on GitHub 😞 

Should this be split across multiple PRs or are either of you willing to review locally (ignoring vendor/ changes?)

**Comment by mattdurham** [comment]
I can, golang will let me exclude vendor

**Review by rlankfo — CHANGES_REQUESTED** [review]
LGTM but I'm not real familiar with all of this code 🤷

I reviewed the non-vendor changes however and didn't see anything obvious.

**Review by rlankfo — APPROVED** [review]

File: example/k3d/scripts/smoke-test.bash [hunk 1]
```diff
@@ -35,15 +35,16 @@ set -euo pipefail
 # Constants
 ROOT=$(git rev-parse --show-toplevel)
 K3D_CLUSTER_NAME="agent-smoke-test"
-MUTATION_FREQUENCY="5m"
-CHAOS_FREQUENCY="30m"
```
Comment (mattdurham) [inline example/k3d/scripts/smoke-test.bash:39]: Is there a reason we went away from the human-readable format for the durations?

**Review by mattdurham — APPROVED** [review]
Reviewed, one comment on human readable names and likely needs merging the changes from main but looks good.

File: example/k3d/scripts/smoke-test.bash (hunk 1 above)
Comment (rfratto) [reply example/k3d/scripts/smoke-test.bash:39]: BSD sleep doesn't support the nice format 😞

**Comment by rfratto** [comment]
I honestly don't know if this PR is still good. I'd rather close this and re-do it again later if we need it.

**Pull request closed without merging**